subprojects/core-api/src/main/java/org/gradle/api/tasks/TaskInputs.java

```diff
@@ -75,7 +75,10 @@
      * Returns the set of input properties for this task.
      *
      * @return The properties.
+     *
+     * @deprecated Use {@link #property(String, Object)} and {@link #properties(Map)} to declare input properties.
```

Comment:
So it was possible to modify the returned collection. Ouch...
I hope no one was actually adding runtime properties by calling `getProperties()`, but I expect to find a place where this was done.

Reply:
It wasn't: we returned a new map every time already.